ineffassign.go

```diff
@@ -12,26 +12,32 @@ import (
 	"strings"
 )
 
-var (
-	root            string
-	dontRecurseFlag = flag.Bool("n", false, "don't recursively check paths")
-)
+var dontRecurseFlag = flag.Bool("n", false, "don't recursively check paths")
 
 func main() {
 	flag.Parse()
-	if len(flag.Args()) != 1 {
+
+	if len(flag.Args()) == 0 {
 		fmt.Println("missing argument: filepath")
-		return
+		os.Exit(3)
 	}
 
-	var err error
-	root, err = filepath.Abs(flag.Arg(0))
-	if err != nil {
-		fmt.Printf("Error finding absolute path :%s", err)
-		return
+	lintFailed := false
+	for _, path := range flag.Args() {
+		root, err := filepath.Abs(path)
+		if err != nil {
+			fmt.Printf("Error finding absolute path: %s", err)
+			os.Exit(3)
+		}
+		lintFailed = walkPath(root) || lintFailed
```

Comment:
An explicit `if walkPath(root) { lintFailed = true }` would be more readable than short-circuit `||`.